core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -486,6 +504,7 @@ private[spark] class BlockManagerInfo(
           blockId, blockManagerId.hostPort,
           Utils.bytesToString(blockStatus.externalBlockStoreSize)))
       }
+      if (!blockId.isBroadcast && blockStatus.isCached) _cachedBlocks += blockId
```

Comment:
wait, why do we add the block here to cached blocks actually? This is the case where we remove the blocks.